app/lib/communicator/communicator.js

```diff
@@ -236,6 +236,7 @@ function getUpdatedRequestOptions( options ) {
     options.headers = options.headers || {};
     options.headers[ 'X-OpenRosa-Version' ] = '1.0';
     options.headers[ 'Date' ] = new Date().toUTCString();
+    options.headers[ 'User-Agent' ] = `Enketo ${config.version}`;
```

Comment:
This means only the Enketo version is included right? Ideally the browser info would still come after. Let’s also follow the convention of separating the product and version by / (eg Enketo/7.4.3)

Reply:
> This means only the Enketo version is included right? Ideally the browser info would still come after.

I didn't realize that was expected. It wasn't previously being sent either, but I expect we can add that. It'll probably be a bit more complex as the originating request context isn't currently available here. There are a few potential approaches to this:

- Pass the originating request from every request handler (functional, but not exactly practical; but I could dig in to see if there's a less "everywhere" way to access it in the current code)
- Some middleware which captures it at the server level (a popular solution, but this can make it harder to reason about types/data flow)
- [Async hooks](https://nodejs.org/api/async_hooks.html), which hold a context until the call stack completes (technically considered "experimental" but I don't think it's much changed for several versions of Node; the API is kind of inscrutable, but there are higher level wrapping libraries available that provide a more reasonable API)

Of those, the second and third are likely the most pragmatic.

I'd also like to consider that these belong in separate headers. Here, Enketo's server is performing the request, and it's operating behalf of a request which may have been initiated by a browser. Perhaps we could put the browser value in `User-Agent` and the Enketo value in `Server`? That doesn't feel quite right either, because Enketo really is the actual user agent performing the request.

> Let’s also follow the convention of separating the product and version by / (eg Enketo/7.4.3)

👍